### PR TITLE
Payment status update for the Object API registration backend

### LINKED ISSUE #713: Implement payment status update for ObjectsAPI registration backend
The Objects API registration backend plugin doesn't have a update_payment_status() implementation.

(it wasn't part of the initial tickets and could be out of sight)

### PR DESCRIPTION
**Pull request by SilviaAmAm**
Fixes #713

### CODE REVIEW

File: src/openforms/registrations/contrib/objects_api/plugin.py
Comment (Viicos) [inline src/openforms/registrations/contrib/objects_api/plugin.py:238]:
```suggestion
                "Skipping payment status update because no template was configured."
```

File: src/openforms/registrations/contrib/objects_api/plugin.py
Comment (Viicos) [inline src/openforms/registrations/contrib/objects_api/plugin.py:249]: This is repeated in the `register_submission` method, I don't know if it's worth moving it in a single method?

**Review by Viicos — APPROVED** [review]

File: src/openforms/registrations/contrib/objects_api/plugin.py
Comment (sergei-maertens) [reply src/openforms/registrations/contrib/objects_api/plugin.py:249]: probably just using a utility function at the module level `def get_payment_context_data(submission: Submission)` would be sufficient :)

File: src/openforms/registrations/contrib/objects_api/plugin.py
Comment (sergei-maertens) [inline src/openforms/registrations/contrib/objects_api/plugin.py:263]: I think it'd be good to have an anti-corruption layer that will allow us to refactor in the future, so wrapping this (here) and in the `register` hook into something with the following signature would be useful:

```py
def render_to_json(template: str, context: dict[str, Any]) -> JSONValue:
    """Render the provided template and context into a JSON serializable object"""
    ...
```

we can then in the future change this implementation with a better solution than Django templates.

File: src/openforms/registrations/contrib/objects_api/plugin.py
Comment (sergei-maertens) [inline src/openforms/registrations/contrib/objects_api/plugin.py:290]: `self` is unused here and nothing about this implementation is really tied to the objects API (it's just currently the only place where we use it), so I'd opt to put this in a simple module-level utility function.

We can then bundle/move it together with the `render_to_json` (and make it part of that function maybe) when we get to it.

File: src/openforms/registrations/contrib/objects_api/plugin.py
Comment (sergei-maertens) [inline src/openforms/registrations/contrib/objects_api/plugin.py:153]: I think we need to properly stringify this before passing to the template engine, since the localization in the templates will cause problems with large numbers. Please add a test case for that situation too (try using number `1_000` but also `3.141592` and see how many decimals are outputted and what the rounding behaviour is).

Because floats are inherently limited, I think the resulting JSON type should be a string where a `.` is used as decimal separator and no grouping occurs, so expected is:

* `1_000` -> `"1000"`
* `3.1415926535` -> `"3.1415926535"`

Unfortunately I don't see a [`format`](https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-validation-00#page-13) for decimal, so maybe a string isn't the correct solution as in theory the object type schema can also specify the value to have a particular precision by using:

```
type: number
multipleOf: 0.01
```

So perhaps accepting the limitations of floats is okay here, since we process the payment already anyway :thinking:

File: src/openforms/registrations/contrib/objects_api/plugin.py
Comment (sergei-maertens) [inline src/openforms/registrations/contrib/objects_api/plugin.py:154]: as remarked above, shouldn't this be an array of IDs instead of making the consumer comma-split it? Let's use the flexibility of JSON we have available.

File: src/openforms/registrations/contrib/objects_api/tests/test_backend.py
Comment (sergei-maertens) [inline src/openforms/registrations/contrib/objects_api/tests/test_backend.py:1264]: This is not reliably machine-parseable, it should probably be:

```suggestion
                "amount": 10.00,
```

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

File: src/openforms/registrations/contrib/objects_api/plugin.py
Comment (SilviaAmAm) [reply src/openforms/registrations/contrib/objects_api/plugin.py:153]: Discussed:
The object type schema should dictate the precision of this field (`type: number, multipleOf: 0.01`). We then send a number in the json and not a string.

File: docs/manual/templates.rst
Comment (sergei-maertens) [inline docs/manual/templates.rst:482]:
```suggestion
In dit sjabloon kunnen alleen de inzendingsvariabelen (``variables.<naam van variabele>``) en de ``payment`` variabele
```

File: docs/manual/templates.rst
Comment (sergei-maertens) [inline docs/manual/templates.rst:509]:
```suggestion
``{{ payment.amount }}``               Bedrag dat betaald moet worden.
```

File: src/openforms/registrations/contrib/objects_api/plugin.py
Comment (sergei-maertens) [inline src/openforms/registrations/contrib/objects_api/plugin.py:229]: isn't it more logical to leave the form builders to specify the `payment` key rather than hardcoding it? IMO this should be part of the template.

File: src/openforms/registrations/contrib/objects_api/tests/test_utils.py
Comment (sergei-maertens) [inline src/openforms/registrations/contrib/objects_api/tests/test_utils.py:101]: I don't understand how this variable results in valid JSON other than pure coincidence. In fact, it may not even be valid JSON due to python strings typically being printend with single quotes rather than double quotes.

File: src/openforms/registrations/contrib/objects_api/utils.py
Comment (sergei-maertens) [inline src/openforms/registrations/contrib/objects_api/utils.py:70]: ah, that explains...

To be honest, I think this is very confusing for the people making templates. Sometimes a variable already outputs, JSON, other times they need to use a special template tag... It's also hard for us to track what is already serialized as JSON and what isn't.

I'd replace this in the default template with a simple for loop:

```django
[{% for order_id in payment.public_order_ids%}"{{ order_id|escapejs }}"{% if not forloop.last %},{% endif %}{% endfor %}
```

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

File: docs/manual/templates.rst
Comment (SilviaAmAm) [reply docs/manual/templates.rst:482]: One day I will get this merging of words right... :stuck_out_tongue:

**Comment by SilviaAmAm** [comment]
Note: should not be merged before the Object API is fixed

**Review by sergei-maertens — APPROVED** [review]
Still waiting for confirmation on the Objects API patch before we merge this.

**Pull request merged**
